packages/context-menu/src/vaadin-contextmenu-items-mixin.js

```diff
@@ -297,8 +298,15 @@ export const ItemsMixin = (superClass) =>
       });
 
       // Listen to the forwarded event from sub-menu.
-      this.addEventListener('item-selected', () => {
-        this.close();
+      this.addEventListener('item-selected', (e) => {
+        const menu = e.target;
+        const selectedItem = e.detail.value;
+
+        if (!!selectedItem.keepOpen && menu.items.includes(selectedItem)) {
+          menu._overlayElement.requestContentUpdate();
```

Comment:
Selecting an item by keyboard doesn't close the overlay (correct), but the menu item loses focus.

Seems to be caused by `requestContentUpdate`.